include/deal.II/fe/fe_enriched.h

```diff
@@ -0,0 +1,545 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__fe_enriched_h
+#define dealii__fe_enriched_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_CXX11
+#include <deal.II/base/thread_management.h>
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_nothing.h>
+#include <deal.II/fe/fe_update_flags.h>
+#include <deal.II/base/function.h>
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/base/quadrature.h>
+
+#include <vector>
+#include <utility>
+#include <numeric>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Implementation of a partition of unity finite element which enriches a standard
+ * finite element with an enrichment function multiplied with another (usually
+ * linear) finite element:
+ * \f[
+ * U(x) = \sum_i N_i(x) U_i + \sum_j N_j(x) \sum_k F_k(x) U_{jk}
+ * \f]
+ * where \f$ N_i(x) \f$ and \f$ N_j(x) \f$ are the underlying finite elements (including
+ * the mapping from the isoparametric element to the real element); \f$ F_k(x) \f$
+ * are the enrichment functions in real space (e.g. \f$ 1/r \f$, \f$ exp(-r) \f$, Heaviside, etc);
+ * \f$ U_i \f$ and \f$ U_{jk} \f$ are the standard and enriched DoFs.
+ *
+ * The simplest constructor of this class takes two finite element objects and an
+ * enrichment function to be used. For example
+ *
+ * @code
+ * FE_Enriched<dim> fe(FE_Q<dim>(1),
+ *                     FE_Q<dim>(1),
+ *                     function)
+ * @endcode
+ *
+ * The \f$ exp(-x) \f$ enrichment leads to the following shape functions on unit
+ * element:
+ * <table>
+ *   <tr>
+ *     <td align="center">
+ *       @image html fe_enriched_1d.png
+ *     </td>
+ *     <td align="center">
+ *       @image html fe_enriched_h-refinement.png
+ *     </td>
+ *   </tr>
+ *   <tr>
+ *     <td align="center">
+ *       1d element, base and partition-of-unity shape functions.
+ *     </td>
+ *     <td align="center">
+ *       enriched shape function resulting from h-refinement of a single element (bottom-left corner).
+ *     </td>
+ *   </tr>
+ * </table>
+ *
+ * <h3>Implementation</h3>
```

Comment:
I think this should be at the very bottom of the documentation, as it is not really of interest to someone who just wants to *use* the class.